nixos/modules/services/development/guix.nix

```diff
@@ -0,0 +1,118 @@
+{config, pkgs, lib, ...}:
+
+with lib;
+
+let
+
+  cfg = config.services.guix;
+
+  guix-binary = pkgs.stdenv.mkDerivation rec
+    { name = "guix-binary-${version}";
+      version = "0.16.0";
+
+      src = pkgs.fetchurl {
+        # TODO: make work on non x86_64 systems
+        url = "https://alpha.gnu.org/gnu/guix/guix-binary-${version}.x86_64-linux.tar.xz";
+        sha256 = "049l0zim30cd0gyly2h3jaw4cshdk78h7xdb9ac173h72i13afbj";
+      };
+      sourceRoot = ".";
+
+      outputs = [ "out" "store" "var" ];
+      phases = [ "unpackPhase" "installPhase" ];
+
+      installPhase = ''
+        # copy the /gnu/store content
+        mkdir -p $store
+        cp -r gnu $store
+
+        # copy /var content
+        mkdir -p $var
+        cp -r var $var
+
+        # link guix binaries
+        mkdir -p $out/bin
+        ln -s /var/guix/profiles/per-user/root/current-guix/bin/guix $out/bin/guix
+        ln -s /var/guix/profiles/per-user/root/current-guix/bin/guix-daemon $out/bin/guix-daemon
+      '';
+    };
+
+  buildGuixUser = i:
+    {
+      "guixbuilder${builtins.toString i}" = {
+        group = "guixbuild";
+        extraGroups = ["guixbuild"];
+        home = "/var/empty";
+        shell = pkgs.nologin;
+        description = "Guix build user ${builtins.toString i}";
+        isSystemUser = true;
+      };
+    };
+
+in {
+
+  options.services.guix = {
+    enable = mkEnableOption "GNU Guix package manager";
+  };
+
+  config = mkIf (cfg.enable) {
+
+    users = {
+      extraUsers = lib.fold (a: b: a // b) {} (builtins.map buildGuixUser (lib.range 1 10));
+      extraGroups.guixbuild = {name = "guixbuild";};
+    };
+
+    environment.systemPackages = [ guix-binary ];
+
+    systemd.services.guix-daemon = {
+      enable = true;
+      description = "Build daemon for GNU Guix";
+      serviceConfig = {
+        ExecStart="/var/guix/profiles/per-user/root/current-guix/bin/guix-daemon --build-users-group=guixbuild";
+        Environment="GUIX_LOCPATH=/var/guix/profiles/per-user/root/guix-profile/lib/locale";
+        RemainAfterExit="yes";
+        StandardOutput="syslog";
```

Comment:
From `systemd.exec(5)`

> Note that the journal daemon is usually configured to forward everything it receives to syslog anyway, in which case this option is no different from journal [the default]

Is there a particular reason why you want to force syslog?

Reply:
No particular reason. I used configurations from the upstream Guix template (https://git.savannah.gnu.org/cgit/guix.git/tree/etc/guix-daemon.service.in).

Maybe it is good to not force it. There is a NixOS options `services.journald.forwardToSyslog`. I don't know exactly what this options does but by not forcing stuff here we might prevent unintended things happening when the option is set manually.

I removed the entry `StandardOutput` and `StandardError`.